src/wallet/api.rs

```diff
@@ -644,24 +662,26 @@ impl Wallet {
         Ok(all_utxos)
     }
 
-    pub(crate) fn get_all_locked_utxo(&self) -> Result<Vec<ListUnspentResultEntry>, WalletError> {
-        let all_utxos = self
-            .rpc
-            .list_unspent(Some(0), Some(9999999), None, None, None)?;
-        Ok(all_utxos)
-    }
+    // pub(crate) fn get_all_locked_utxo(&self) -> Result<Vec<ListUnspentResultEntry>, WalletError> {
+    //     let all_utxos = self
+    //         .rpc
+    //         .list_unspent(Some(0), Some(9999999), None, None, None)?;
+    //     Ok(all_utxos)
+    // }
+
     /// Returns a list all utxos with their spend info tracked by the wallet.
     /// Optionally takes in an Utxo list to reduce RPC calls. If None is given, the
     /// full list of utxo is fetched from core rpc.
     pub fn list_all_utxo_spend_info(
         &self,
-        utxos: Option<&Vec<ListUnspentResultEntry>>,
     ) -> Result<Vec<(ListUnspentResultEntry, UTXOSpendInfo)>, WalletError> {
-        let all_utxos = if let Some(utxos) = utxos {
-            utxos.clone()
-        } else {
-            self.get_all_utxo()?
-        };
+        // Use cached UTXOs for Maker
+        let all_utxos: Vec<ListUnspentResultEntry> = self
+            .store
+            .utxo_cache
+            .values()
+            .map(|(utxo, _spend_info)| utxo.clone())
+            .collect();
 
         let processed_utxos = all_utxos
             .iter()
```

Comment:
We already had the spend info data in the map? Why not use the same data instead of reprcoessing the spend info again? this function can be just the wrapper API over the map to extract all the data.

Reply:
Yes but you don't need the processing logic anymore.. just map it to `(ListUnspentResultEntry, UTXOSpendInfo)`.